server/src/test/java/io/druid/segment/realtime/RealtimeManagerTest.java

```diff
@@ -452,7 +453,12 @@ public int add(InputRow row) throws IndexSizeExceededException
         return -1;
       }
 
-      return sink.add(row);
+      final int numRows = sink.add(row);
+      if (!sink.canAppendRow()) {
```

Comment:
if it is not doing anything, then we can remove it and not complicate life for no reason :)